workspaces/src/operations.rs

```diff
@@ -398,3 +434,61 @@ impl<'a, 'b> CreateAccountTransaction<'a, 'b> {
         })
     }
 }
+
+/// `TransactionStatus` object relating to an [`asynchronous transaction`] on the network.
+/// Used to query into the status of the Transaction for whether it has completed or not.
+///
+/// [`asynchronous transaction`]: https://docs.near.org/api/rpc/transactions#send-transaction-async
+#[must_use]
+pub struct TransactionStatus<'a> {
+    client: &'a Client,
+    sender_id: AccountId,
+    hash: CryptoHash,
+}
+
+impl<'a> TransactionStatus<'a> {
+    pub(crate) fn new(
+        client: &'a Client,
+        id: AccountId,
+        hash: near_primitives::hash::CryptoHash,
+    ) -> Self {
+        Self {
+            client,
+            sender_id: id,
+            hash: CryptoHash(hash.0),
+        }
+    }
+
+    /// Checks the status of the transaction. If an `Err` is returned, then the
+    /// transaction has not completed yet.
+    pub async fn status(&self) -> Result<ExecutionFinalResult> {
```

Comment:
Isn't this misleading that a non-finalized result would be treated as a result? Wouldn't you want something like `Result<Option<..>>` or some other way to indicate the `Poll::NotReady` state essentially?

Reply:
true, we should probably have a different variant than just `Result`. I was just parroting what we had in the client side, but definitely better for users if we had a custom enum that portrayed this poll-like state